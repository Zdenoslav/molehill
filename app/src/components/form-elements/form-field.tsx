import React from 'react';
import { BaseFieldProps, Field, WrappedFieldProps } from 'redux-form'
import styled from 'styled-components';

const FieldContainer = styled.div`
  margin: 8px 0;

`;

type Props = BaseFieldProps & Partial<WrappedFieldProps> & Partial<HTMLInputElement>;

const FormField: React.SFC<Props> = (props) => {

  return (
    <FieldContainer>
      <Field {...props} />
    </FieldContainer>
  );
}

export default FormField;