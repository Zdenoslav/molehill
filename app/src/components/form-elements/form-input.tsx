import React from 'react';
import { WrappedFieldProps } from 'redux-form';

import styled from 'app/components/styleguide';
import { s2 } from '../styleguide/spacing';

import {Body} from 'app/components/styleguide/text';

const StyledInput = styled.input`
  border-radius: 7px;
  padding: 8px;
  outline: none;
  border: 1px solid ${props => props.theme.border.default};
  font-size: 16px;
  width: 100%;
  transition: 0.3s all;

  &:focus {
    outline: none;
    border: 1px solid ${props => props.theme.border.focus};
    box-shadow: 0 1px 2px ${props => props.theme.shadow};
  }
`;

const InputMessage = styled(Body)`
  color: ${props => props.theme.error};
`;

const FormInput: React.SFC<WrappedFieldProps> = (props) => {
  const {
    input,
    meta: { touched, error, warning },
    ...rest
  } = props;

  let inputMessage: string | undefined;
  inputMessage = warning && warning;
  inputMessage = error && error;

  return (
    <>
      <StyledInput
        {...input}
        {...rest}
      />
      {touched && <InputMessage marginTop={s2}>{inputMessage}</InputMessage>}
    </>
  );
};

export default FormInput;
