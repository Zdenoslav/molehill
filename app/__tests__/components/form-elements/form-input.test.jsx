import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {FormInput, StyledInput, InputContainer, ClearButton} from 'app/components/form-elements/form-input';
import Theme from 'app/components/styleguide/theme';

describe('FormInput', () => {

  it('should render FormInput', () => {
    const wrapper = shallow(<FormInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render StyledInput', () => {
    const wrapper = renderer.create(<StyledInput theme={Theme} />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('border', `1px solid ${Theme.border.default}`);
  });

  it('should render InputContainer', () => {
    const wrapper = renderer.create(<InputContainer />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('position', 'relative');
  });

  it('should render ClearButton', () => {
    const wrapper = renderer.create(<ClearButton />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('position', 'absolute');
  });

  it('should clear input on ClearButton clicked', () => {
    const clearInput = jest.fn();
    const wrapper = shallow(<ClearButton onClick={clearInput} />);
    wrapper.simulate('click');
    expect(clearInput).toHaveBeenCalled();
  });

});
