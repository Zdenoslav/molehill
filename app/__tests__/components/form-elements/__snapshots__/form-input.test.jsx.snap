// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render ClearButton 1`] = `
.c0 {
  width: auto;
  font-weight: 400;
  opacity: 1;
  word-break: unset;
  font-size: 20px;
  position: absolute;
  right: 8px;
  top: 4px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<p
  className="c0"
/>
`;

exports[`FormInput should render FormInput 1`] = `
<styled.div>
  <styled.input />
</styled.div>
`;

exports[`FormInput should render InputContainer 1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`FormInput should render StyledInput 1`] = `
.c0 {
  border-radius: 7px;
  padding: 8px;
  outline: none;
  border: 1px solid #e6e6e6;
  font-size: 14px;
  width: 100%;
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
}

.c0:focus {
  outline: none;
  border: 1px solid #bababa;
  box-shadow: 0 1px 2px rgba(160,160,160,0.35);
}

<input
  className="c0"
/>
`;
